initial commit (#1)
qbeyond/terraform-azurerm-alz-rbac: README.md

# Module

----

> This terraform module creates Azure Ad groups and role assignments for management groups and subscriptions. It is specifically designed to work in the governance module that is created by the Prerequisites oneshot deployment module

Reviewer: ```suggestion
This terraform module creates Azure AD groups and role assignments for management groups and subscriptions. It is specifically designed to work in the governance module that is created by the prerequisites oneshot deployment.
```

The module is designed to create default groups (Owner, Contributor, and Reader) and assignments for the specified subscriptions and management groups. The module automatically generates the name of the group by prefixing it with `SUB_` for subscriptions and `AMG_` for management groups and adds the role as a suffix.

> The Reader and Owner groups created for subscriptions and management groups are automatically enabled for Privileged Identity Management (PIM)

Reviewer: ```suggestion
The Contributor and Owner groups created for subscriptions and management groups are automatically enabled for Privileged Identity Management (PIM)
```

> The `custom_assignments` input variable allows you to define custom group assignments, including the ability to enable Privileged Identity Management (PIM) for the assignments.

Reviewer: ```suggestion
The `custom_assignments` input variable allows you to define custom groups and assignments, including the ability to enable Privileged Identity Management (PIM) for the groups.
```

> TODO: there is currently no stage variable so this module will run into errors when you have no aad premium license.also in certain development enviroments you have to add the skip `skip_provider_registration` flag.

Reviewer: ```suggestion
TODO: there is currently no stage variable so this module will run into errors when you have no AAD premium license.
```

> **Important**: The `management_groups` variable reuses the structure of the `custom_landing_zone` variable from the [caf module](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/wiki/%5BExamples%5D-Deploy-Custom-Landing-Zone-Archetypes)

Reviewer: ```suggestion
**Important**: The `management_groups` variable reuses the structure of the `custom_landing_zone` variable from the [CAF module](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/wiki/%5BExamples%5D-Deploy-Custom-Landing-Zone-Archetypes)
```
